lib/rules/no-shadow-route-definition.js

```diff
@@ -71,11 +71,9 @@ module.exports = {
           return;
         }
 
-        if (
-          routeMap.has(routeInfo.route.fullPathWithGenericParams) &&
-          !isNestedRouteWithSamePath(routeInfo)
-        ) {
-          const existingRouteInfo = routeMap.get(routeInfo.route.fullPathWithGenericParams);
+        const routeLookupKey = `${routeInfo.route.blockStatementsTreePrefix}::${routeInfo.route.fullPathWithGenericParams}`;
```

Comment:
For routes that are not within if else blocks, key will look like "::<path>" - we could set it "global::<path>" for easier understanding what's the contents of route map. 🤷‍♂️